src/main/java/school/hei/haapi/service/FeeService.java

```diff
@@ -11,7 +13,36 @@
 
   private final FeeRepository feeRepository;
 
-  public Fee getByUserIdAndId(String userId, String feeId) {
-    return feeRepository.getByUserIdAndId(userId, feeId);
+  public Fee getByStudentIdAndFeeId(String userId, String feeId) {
+    return feeRepository.getByStudentIdAndId(userId, feeId);
+  }
+
+  public List<Fee> saveAll(String studentId, List<Fee> fees) {
+    for (Fee f : fees) {
+      checkStudentFee(studentId, f);
```

Comment:
validity check is always done as first statement.

before the for-loop, call `checkCreateStudentFees(studentId, fees)`. And validate all fees before throwing an exception that list all the errors (like what HibernateValidator does)